common/event_emitter.go

```diff
@@ -97,8 +98,12 @@ type NavigationEvent struct {
 }
 
 type eventHandler struct {
-	ctx context.Context
-	ch  chan Event
+	ctx           context.Context
+	ch            chan Event
+	queueMutex    *sync.Mutex
+	queue         []Event
+	curQueueMutex *sync.Mutex
+	curQueue      []Event
```

Comment:
Since `eventHandler` is used as a pointer, we can now drop the sync mutex pointers. I made changes to the relevant parts in the code to be able to use this.

```suggestion
	ctx           context.Context
	ch            chan Event
	queueMutex    sync.Mutex
	queue         []Event
	curQueueMutex sync.Mutex
	curQueue      []Event
```